bpython/curtsiesfrontend/repl.py

```diff
@@ -1847,12 +1837,20 @@ def take_back_buffer_line(self):
             self.cursor_offset = len(self.current_line)
         self.display_buffer.pop()
         self.buffer.pop()
-        self.history.pop()
+        undone_input = self.history.pop()
+
+        # Pop all outputs leading up to the undone input
```

Comment:
We shouldn't need this for `take_back_buffer_line()`, the buffer is the current (as yet unexecuted) possibly-multiline input statement, so removing a line will not affect output